jerry-core/ecma/builtin-objects/ecma-builtin-string-prototype.cpp

```diff
@@ -1678,12 +1799,277 @@ ecma_builtin_string_prototype_object_slice (ecma_value_t this_arg, /**< this arg
  */
 static ecma_completion_value_t
 ecma_builtin_string_prototype_object_split (ecma_value_t this_arg, /**< this argument */
-                                            ecma_value_t arg1, /**< routine's first argument */
-                                            ecma_value_t arg2) /**< routine's second argument */
+                                            ecma_value_t arg1, /**< separator */
+                                            ecma_value_t arg2) /**< limit */
 {
-  ECMA_BUILTIN_CP_UNIMPLEMENTED (this_arg, arg1, arg2);
+  ecma_completion_value_t ret_value = ecma_make_empty_completion_value ();
+
+  /* 1. */
+  ECMA_TRY_CATCH (this_check_coercible_val,
+                  ecma_op_check_object_coercible (this_arg),
+                  ret_value);
+
+  /* 2. */
+  ECMA_TRY_CATCH (this_to_string_val,
+                  ecma_op_to_string (this_arg),
+                  ret_value);
+
+  /* 3. */
+  ecma_completion_value_t new_array = ecma_op_create_array_object (0, 0, false);
+
+  /* 5. */
+  ecma_length_t limit = 0;
+
+  if (ecma_is_value_undefined (arg2))
+  {
+    limit = (ecma_length_t) -1;
```

Comment:
Please, use `(uint32_t) -1` for the cases, as `ecma_length_t` is not guaranteed to always be `uint32_t`.